spe.py

```diff
@@ -0,0 +1,318 @@
+#!/usr/bin/env python
+'''
+Do the Single Photoelectron anaylsis
+
+Usage:
+  spe.py [options] <files>...
+
+Options:
+  -h --help     Show this screen.
+'''
+
+from docopt import docopt
+from digicampipe.io.event_stream import event_stream
+import numpy as np
+import matplotlib.pyplot as plt
+from histogram.histogram import Histogram1D
+from tqdm import tqdm
+import peakutils
+from  scipy.signal import find_peaks_cwt
+from digicampipe.utils.utils import integrate
+from scipy import ndimage
+import scipy
+from iminuit import Minuit
+from digicampipe.io.containers import CalibrationContainer
+from digicampipe.utils.utils import filter_template
+
+
+def calibration_event_stream(path, telescope_id, max_events):
+
+    container = CalibrationContainer()
+
+    for event in event_stream(path, max_events=max_events):
+
+        container.adc_samples = event.r0.tel[telescope_id].adc_samples
+        container.n_pixels = container.adc_samples.shape[0]
+
+        yield container
+
+
+def compute_event_stats(events):
+
+    for count, event in enumerate(events):
+
+        if count == 0:
+
+            n_pixels = event.n_pixels
+            adc_histo = Histogram1D(data_shape=(n_pixels, ), bin_edges=np.arange(0, 4095, 1), axis_name='[LSB]')
+
+        adc_histo.fill(event.adc_samples)
+
+    mode = adc_histo.mode()
+    mean = adc_histo.mean()
+    std = adc_histo.std()
+    max = adc_histo.max
```

Comment:
I personally would return the `adc_histo` and let the user of this function decide which field of the histogram she might find most interesting.